packages/terra-application-links/src/tabs/ApplicationTabs.jsx

```diff
@@ -0,0 +1,168 @@
+   * within a StaticRouter.
+   */
+  staticContext: PropTypes.object,
+};
+
+const defaultProps = {
+  links: [],
+};
+
+class ApplicationTabs extends React.Component {
+  constructor(props) {
+    super(props);
+    this.setContainerNode = this.setContainerNode.bind(this);
+    this.handleResize = this.handleResize.bind(this);
+    this.resetCalculations();
+  }
+
+  componentDidMount() {
+    this.resizeObserver = new ResizeObserver(() => {
+      // Resetting the state so that all elements will be rendered face-up for width calculations
+      if (this.hiddenStartIndex !== -1 || this.menuHidden || !this.isCalculating) {
+        this.resetCalculations();
+        this.forceUpdate();
+      }
+    });
+    this.resizeObserver.observe(this.container);
+    this.handleResize(this.container.clientWidth);
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (this.props.location !== newProps.location) {
+      this.resetCalculations();
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.isCalculating) {
+      this.handleResize(this.container.clientWidth);
+    }
+  }
+
+  componentWillUnmount() {
+    this.resizeObserver.disconnect(this.container);
+    this.container = null;
+  }
+
+  setContainerNode(node) {
+    if (node === null) { return; } // Ref callbacks happen on mount and unmount, element will be null on unmount
+    this.container = node;
+  }
+
+  resetCalculations() {
+    this.hiddenStartIndex = -1;
+    this.menuHidden = false;
+    this.isCalculating = true;
+  }
+
+  handleResize(width) {
+    // Calculate hide index
+    const childrenCount = this.props.links.length;
+    const tabWidth = childrenCount > 1 ? this.container.children[0].getBoundingClientRect().width : 0;
+    const availableWidth = width - tabWidth;
+
+    let newHideIndex = childrenCount;
+    let calcMinWidth = 0;
+    let isMenuHidden = true;
+    for (let i = 0; i < childrenCount; i += 1) {
+      calcMinWidth += tabWidth;
+      if (calcMinWidth > availableWidth && !(i === childrenCount - 1 && calcMinWidth <= width)) {
+        newHideIndex = i;
+        isMenuHidden = false;
+        break;
+      }
+    }
+
+    if (this.hiddenStartIndex !== newHideIndex) {
+      this.hiddenStartIndex = newHideIndex;
+      this.menuHidden = isMenuHidden;
+      this.isCalculating = false;
+      this.forceUpdate();
+    }
+  }
+
+  render() {
+    const {
+      links,
+      location,
+      match,
+      history,
+      staticContext,
+      ...customProps
+    } = this.props;
+
+    const visibleChildren = [];
+    const hiddenChildren = [];
+
+    links.forEach((link, index) => {
+      if (this.hiddenStartIndex < 0) {
+        visibleChildren.push(<ApplicationTab id={link.id} path={link.path} text={link.text} key={link.path} />);
+        hiddenChildren.push(<ApplicationTab id={link.id} path={link.path} text={link.text} key={link.path} isHidden />);
+      } else if (index < this.hiddenStartIndex || this.hiddenStartIndex < 0) {
```

Comment:
in this else-if, I dont think this condition ` || this.hiddenStartIndex < 0` will ever be hit given the initial if condition